src/mlpack/methods/sparse_svm/sparse_svm_function.hpp

```diff
@@ -57,6 +79,41 @@ class SparseSVMFunction
                 GradType& gradient,
                 const size_t batchSize = 1);
 
+  /**
+   * Evaluate the gradient of the hinge loss function, following
+   * the SparseFunctionType requirements on the Gradient function
+   * followed by evaluation of the hinge loss function on all the
+   * datapoints
+   *
+   * @tparam GradType
+   * @param parameters The parameters of the SVM.
+   * @param gradient Sparse matrix to output the gradient into.
+   * @return The value of the loss function at the given parameters.
+   */
+  template <typename GradType>
+  double EvaluateWithGradient(const arma::mat& parameters,
+                              GradType& gradient) const;
+
+  /**
+   * Evaluate the gradient of the hinge loss function, following
+   * the SparseFunctionType requirements on the Gradient function
+   * followed by evaluation of the hinge loss function on the specified
+   * datapoints.
+   *
+   * @tparam GradType
```

Comment:
Oops, I think you forgot documentation for this.  I guess `Type of the gradient matrix` could work, for instance.